src/syntax_analyzer.cpp

```diff
@@ -1 +1,1511 @@
+  } else {
+    this->ValidateIsWord(this->peek_first_token());
+    std::string action_str =
+        std::dynamic_pointer_cast<StringNode>(action)->get_data();
+    if (action_str == "ADD") {
+      argument = this->GetTableDefinition();
+    } else if (action_str == "DROP") {
+      argument = this->GetDropListDefinition();
+    } else {
+      LOG(ERROR, "invalid alter table query: incorrect action \'"
+          << action_str << "\'");
+      return node;
+    }
+    SyntaxAnalyzer::MakeKinship(action, argument);
+  }
+
+  return node;
+}
+std::shared_ptr<Node> SyntaxAnalyzer::GetDropDatabaseSt() {
+  std::shared_ptr<Node> node, database_name, separator;
+  node = std::dynamic_pointer_cast<Node>(std::make_shared<ServiceNode>());
+  node->set_st_type(StatementType::dropDatabaseStatement);
+
+  // First database_name
+  if (tokens_array_.empty()) {
+    LOG(ERROR, "databaseName is missed");
+  } else {
+    this->ValidateIsWord(this->peek_first_token());
+    database_name = this->GetName();
+    SyntaxAnalyzer::MakeKinship(node, database_name);
+  }
+
+  if (!tokens_array_.empty()) {
+    if (SyntaxAnalyzer::IsComma(this->peek_first_token())) {
+      separator = this->GetListOf(StatementType::name);
+      SyntaxAnalyzer::MakeKinship(node, separator);
+    }
+  }
+
+  return node;
+}
+std::shared_ptr<Node> SyntaxAnalyzer::GetDropTableSt() {
+  std::shared_ptr<Node> node, table_name, separator;
+  node = std::dynamic_pointer_cast<Node>(std::make_shared<ServiceNode>());
+  node->set_st_type(StatementType::dropTableStatement);
+
+  // First tableName
+  if (tokens_array_.empty()) {
+    LOG(ERROR, "table name is missed");
+  } else {
+    this->ValidateIsWord(peek_first_token());
+    table_name = this->GetName();
+    SyntaxAnalyzer::MakeKinship(node, table_name);
+  }
+
+  if (!tokens_array_.empty()) {
+    if (SyntaxAnalyzer::IsComma(this->peek_first_token())) {
+      separator = this->GetListOf(StatementType::name);
+      SyntaxAnalyzer::MakeKinship(node, separator);
+    }
+  }
+
+  return node;
+}
+
+// DDL Basic Statements
+
+std::shared_ptr<Node> SyntaxAnalyzer::GetTableDefinition() {
+  std::shared_ptr<Node> node, argument, separator, next_arguments;
+  node = std::dynamic_pointer_cast<Node>(std::make_shared<ServiceNode>());
+  node->set_st_type(StatementType::tableDefinition);
+
+  std::string key_word =
+      std::dynamic_pointer_cast<StringNode>(
+          this->peek_first_token())->get_data();
+  bool is_tableConstraint =
+      key_word == "CONSTRAINT"
+          || key_word == "PRIMARY"
+          || key_word == "FOREIGN";
+
+  if (is_tableConstraint) {
+    argument = this->GetTableConstraint();
+  } else {
+    argument = this->GetColumnDefinition();
+  }
+  SyntaxAnalyzer::MakeKinship(node, argument);
+
+  if (!tokens_array_.empty()) {
+    if (SyntaxAnalyzer::IsComma(this->peek_first_token())) {
+      separator = this->GetTableDefinitionList();
+      SyntaxAnalyzer::MakeKinship(node, separator);
+    }
+  }

```

Comment:
Duplicating code from a function `GetTableDefinitionList()`. Maybe, to get rid of duplicating, you should instead of this code call `GetTableDefinitionList()`, but for this you should add some checks in this function.

Reply:
fixed